### PR TITLE
Merge data layer for bulk pricing epic.

### LINKED ISSUE #10560: Create data layer for bulk price information.
**Is your feature request related to a problem? Please describe.**
Create data layer for obtaining bulk price information for a product using orgProducts endpoint.

### PR DESCRIPTION
**Pull request by ren-tham**
Introducing data layer to bulk-pricing to epic.

Closes #10560

### CODE REVIEW

File: feature-libs/product/bulk-pricing/components/bulk-pricing-table/bulk-pricing-table.component.ts [hunk 1]
```diff
@@ -1,12 +1,43 @@
...
-  ngOnInit(): void {}
+  constructor(
+    private routingService: RoutingService,
+    private bulkPrices: BulkPricesService
+  ) {
+    this.testString = 'bulk-pricing-table works!';
+  }
+
+  ngOnInit(): void {
+    this.dataStream = this.getPrices().subscribe((p) => {
```
Comment (mszczepaniak) [inline feature-libs/product/bulk-pricing/components/bulk-pricing-table/bulk-pricing-table.component.ts]: let's rename it to this.prices$

File: feature-libs/product/bulk-pricing/components/bulk-pricing-table/bulk-pricing-table.component.ts [hunk 2]
```diff
@@ -1,12 +1,43 @@
...
+      this.pricingTiers = p;
+      console.log({ p });
```
Comment (mszczepaniak) [inline feature-libs/product/bulk-pricing/components/bulk-pricing-table/bulk-pricing-table.component.ts]: let's remove the log

File: feature-libs/product/bulk-pricing/components/bulk-pricing-table/bulk-pricing-table.component.ts [hunk 3]
```diff
@@ -1,12 +1,43 @@
...
+    });
+  }
+
+  getPrices(): Observable<BulkPrice[]> {
+    return this.routingService.getRouterState().pipe(
```
Comment (mszczepaniak) [inline feature-libs/product/bulk-pricing/components/bulk-pricing-table/bulk-pricing-table.component.ts:35]: could we think of another way of obtaining the product code without the usage of the router?

File: feature-libs/product/bulk-pricing/components/bulk-pricing-table/bulk-pricing-table.component.ts [hunk 4]
```diff
@@ -1,12 +1,43 @@
...
+      switchMap((state) => {
+        const productCode = state.state.params['productCode'];
```
Comment (mszczepaniak) [inline feature-libs/product/bulk-pricing/components/bulk-pricing-table/bulk-pricing-table.component.ts]: let's extract the hardcoded string to a const

File: feature-libs/product/bulk-pricing/core/services/bulk-prices.service.ts [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+      value: priceTier.value,
+      formattedDiscount: '',
+      discount: 0,
+    };
+
+    this.calculateDiscount(bulkPrice, basePrice);
+    return bulkPrice;
+  }
+
+  calculateDiscount(bulkPrice: BulkPrice, basePrice: number): void {
```
Comment (mszczepaniak) [inline feature-libs/product/bulk-pricing/core/services/bulk-prices.service.ts]:
```suggestion
  private calculateDiscount(bulkPrice: BulkPrice, basePrice: number): void {
```

File: feature-libs/product/bulk-pricing/core/services/bulk-prices.service.ts [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+      content = content.volumePrices;
+      for (let i = 0; i < content.length; i++) {
+        const tierPrice = this.parsePrice(content[i], basePrice);
+        bulkPrices.push(tierPrice);
+      }
+    }
+    return bulkPrices;
+  }
+
+  parsePrice(priceTier: any, basePrice): BulkPrice {
```
Comment (mszczepaniak) [inline feature-libs/product/bulk-pricing/core/services/bulk-prices.service.ts]:
```suggestion
  private parsePrice(priceTier: any, basePrice): BulkPrice {
```

File: feature-libs/product/bulk-pricing/core/services/bulk-prices.service.ts [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+
+  getBulkPrices(productCode: string): Observable<BulkPrice[]> {
+    return this.productService.get(productCode, this.PRODUCT_SCOPE).pipe(
+      switchMap((p) => {
+        return of(this.convert(p));
+      })
+    );
+  }
+
+  convert(content: any): BulkPrice[] {
```
Comment (mszczepaniak) [inline feature-libs/product/bulk-pricing/core/services/bulk-prices.service.ts]:
```suggestion
  private convert(content: any): BulkPrice[] {
```

File: feature-libs/product/bulk-pricing/components/bulk-pricing-table/bulk-pricing-table.component.ts (hunk 2 above)
Comment (ren-tham) [reply feature-libs/product/bulk-pricing/components/bulk-pricing-table/bulk-pricing-table.component.ts]: Done

File: feature-libs/product/bulk-pricing/core/services/bulk-prices.service.ts (hunk 7 above)
Comment (ren-tham) [reply feature-libs/product/bulk-pricing/core/services/bulk-prices.service.ts]: Done

File: feature-libs/product/bulk-pricing/core/services/bulk-prices.service.ts (hunk 6 above)
Comment (ren-tham) [reply feature-libs/product/bulk-pricing/core/services/bulk-prices.service.ts]: Done

File: feature-libs/product/bulk-pricing/core/services/bulk-prices.service.ts (hunk 5 above)
Comment (ren-tham) [reply feature-libs/product/bulk-pricing/core/services/bulk-prices.service.ts]: Done

File: feature-libs/product/bulk-pricing/components/bulk-pricing-table/bulk-pricing-table.component.ts (hunk 1 above)
Comment (ren-tham) [reply feature-libs/product/bulk-pricing/components/bulk-pricing-table/bulk-pricing-table.component.ts]: Done

File: feature-libs/product/bulk-pricing/components/bulk-pricing-table/bulk-pricing-table.component.ts (hunk 4 above)
Comment (ren-tham) [reply feature-libs/product/bulk-pricing/components/bulk-pricing-table/bulk-pricing-table.component.ts]: Done

File: feature-libs/product/bulk-pricing/components/bulk-pricing-table/bulk-pricing-table.component.ts (hunk 3 above)
Comment (ren-tham) [reply feature-libs/product/bulk-pricing/components/bulk-pricing-table/bulk-pricing-table.component.ts:35]: Investigating, other options

File: feature-libs/product/bulk-pricing/core/services/bulk-prices.service.ts [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+      content = content.volumePrices;
+      for (let i = 0; i < content.length; i++) {
+        const tierPrice = this.parsePrice(content[i], basePrice);
+        bulkPrices.push(tierPrice);
+      }
+    }
+    return bulkPrices;
+  }
+
+  private parsePrice(priceTier: any, basePrice): BulkPrice {
```
Comment (mszczepaniak) [inline feature-libs/product/bulk-pricing/core/services/bulk-prices.service.ts]: Can we add the type for the priceTier?

File: feature-libs/product/bulk-pricing/core/services/bulk-prices.service.ts [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+
+  getBulkPrices(productCode: string): Observable<BulkPrice[]> {
+    return this.productService.get(productCode, this.PRODUCT_SCOPE).pipe(
+      switchMap((p) => {
+        return of(this.convert(p));
+      })
+    );
+  }
+
+  private convert(content: any): BulkPrice[] {
```
Comment (mszczepaniak) [inline feature-libs/product/bulk-pricing/core/services/bulk-prices.service.ts]: Can we add the type for the content?

File: feature-libs/product/bulk-pricing/core/services/bulk-prices.service.ts (hunk 8 above)
Comment (ren-tham) [reply feature-libs/product/bulk-pricing/core/services/bulk-prices.service.ts]: done

File: feature-libs/product/bulk-pricing/core/services/bulk-prices.service.ts (hunk 9 above)
Comment (ren-tham) [reply feature-libs/product/bulk-pricing/core/services/bulk-prices.service.ts]: done

**Pull request merged**